### PR TITLE
Feature/auction burn section

### PR DESCRIPTION
**Pull request by Chef-Cheems**
Farm auction cake burn section + some tiny fixes

### CODE REVIEW

**Review by hachiojidev — APPROVED** [review]

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    if (!burnedCakeAmount) {
+      fetchBurnedCakeAmount()
+    }
+  }, [burnedCakeAmount, farmAuctionContract])
+  return (
+    <Flex flexDirection={['column-reverse', null, 'row']}>
+      <Flex flexDirection="column" flex="2">
+        {burnedCakeAmount !== null ? (
+          <Text fontSize="64px" bold>
+            <CountUp start={0} end={burnedCakeAmount} suffix=" CAKE" decimals={0} duration={1} separator="," />
```
Comment (ChefHutch) [inline src/views/FarmAuction/components/AuctionCakeBurn.tsx:37]: y u no like `<Balance>`? 😢

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const AuctionCakeBurn: React.FC = () => {
+  const [burnedCakeAmount, setBurnedCakeAmount] = useState(null)
+  const { t } = useTranslation()
+  const farmAuctionContract = useFarmAuctionContract()
+  const cakePriceBusd = usePriceCakeBusd()
+
+  const burnedAmountAsUSD = cakePriceBusd.times(burnedCakeAmount).toNumber()
+
+  useEffect(() => {
+    const fetchBurnedCakeAmount = async () => {
```
Comment (ChefHutch) [inline src/views/FarmAuction/components/AuctionCakeBurn.tsx:19]: Not sure how far down the page this appears but could be worth using the intersectionObserver? 🗑️ this suggestion if it's near the top of the page

**Review by ChefHutch — APPROVED** [review]

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          {t('through community auctions so far!')}
+        </Text>
+        {!Number.isNaN(burnedAmountAsUSD) ? (
+          <Text color="textSubtle">~${burnedAmountAsUSD.toLocaleString('en', { maximumFractionDigits: 0 })}</Text>
+        ) : (
+          <Skeleton width="128px" />
+        )}
+      </Flex>
+      <Flex flex="1" alignSelf="center">
+        <img width="350px" height="320px" src="/images/burnt-cake.png" alt="Burnt CAKE" />
```
Comment (RabbitDoge) [inline src/views/FarmAuction/components/AuctionCakeBurn.tsx:55]: Use the Image component from the uikit to benefit from lazyloading

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx (hunk 2 above)
Comment (Chef-Cheems) [reply src/views/FarmAuction/components/AuctionCakeBurn.tsx:19]: You need to scroll to it, but it fetches just one thing once and is super fast, don't think it worth the hassle

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import { getBalanceNumber } from 'utils/formatBalance'
+import { ethersToBigNumber } from 'utils/bigNumber'
+
+const AuctionCakeBurn: React.FC = () => {
+  const [burnedCakeAmount, setBurnedCakeAmount] = useState(null)
+  const { t } = useTranslation()
+  const farmAuctionContract = useFarmAuctionContract()
+  const cakePriceBusd = usePriceCakeBusd()
+
+  const burnedAmountAsUSD = cakePriceBusd.times(burnedCakeAmount).toNumber()
```
Comment (RabbitDoge) [inline src/views/FarmAuction/components/AuctionCakeBurn.tsx:16]: We should try to stick to the rule of keeping a BN representation until the number is displayed in a component.
Which would involve removing `.toNumber()` here , and replace line 48 by `isNan()` function from the BN lib.

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        ) : (
+          <Skeleton width="256px" height="64px" />
+        )}
+        <Text textTransform="uppercase" fontSize="64px" bold color="secondary">
+          {t('Burned')}
+        </Text>
+        <Text fontSize="24px" bold>
+          {t('through community auctions so far!')}
+        </Text>
+        {!Number.isNaN(burnedAmountAsUSD) ? (
```
Comment (RabbitDoge) [inline src/views/FarmAuction/components/AuctionCakeBurn.tsx:48]: keep BN representation and use http://mikemcl.github.io/bignumber.js/#isNaN (see comment above)

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx (hunk 1 above)
Comment (Chef-Cheems) [reply src/views/FarmAuction/components/AuctionCakeBurn.tsx]: Fixed in 26e2996edec437674209bbe3ecae5a1bf81e957b

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx (hunk 4 above)
Comment (Chef-Cheems) [reply src/views/FarmAuction/components/AuctionCakeBurn.tsx]: Fixed in 6ca621e8b26d77188c952317d2ce3e3b5bcde601

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx (hunk 5 above)
Comment (Chef-Cheems) [reply src/views/FarmAuction/components/AuctionCakeBurn.tsx]: Fixed in 6ca621e8b26d77188c952317d2ce3e3b5bcde601

File: src/views/FarmAuction/components/AuctionCakeBurn.tsx (hunk 3 above)
Comment (Chef-Cheems) [reply src/views/FarmAuction/components/AuctionCakeBurn.tsx]: Fixed in ea49d7601700caa3788620c621210067632e17a1

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**